plugins/modules/ntnx_volume_groups.py

```diff
@@ -0,0 +1,413 @@
+    type: bool
+    default: false
+  target_password:
+    description: CHAP secret
+    type: str
+extends_documentation_fragment:
+      - nutanix.ncp.ntnx_credentials
+      - nutanix.ncp.ntnx_operations
+author:
+  - Prem Karat (@premkarat)
+  - Gevorg Khachatryan (@Gevorg-Khachatryan-97)
+  - Alaa Bishtawi (@alaa-bish)
+"""
+
+EXAMPLES = r"""
+
+"""
+
+RETURN = r"""
+
+"""
+
+from ..module_utils.base_module import BaseModule  # noqa: E402
+from ..module_utils.prism.iscsi_clients import Clients  # noqa: E402
+from ..module_utils.prism.tasks import Task  # noqa: E402
+from ..module_utils.prism.vdisks import VDisks  # noqa: E402
+from ..module_utils.prism.volume_groups import VolumeGroup  # noqa: E402
+from ..module_utils.utils import remove_param_with_none_value  # noqa: E402
+
+
+def get_module_spec():
+    mutually_exclusive = [("name", "uuid")]
+
+    entity_by_spec = dict(name=dict(type="str"), uuid=dict(type="str"))
+
+    disk_spec = dict(
+        size_gb=dict(type="int"),
+        storage_container=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive
+        ),
+    )
+
+    client_spec = dict(
+        uuid=dict(type="str"),
+        iscsi_iqn=dict(type="str"),
+        iscsi_ip=dict(type="str"),
+        client_password=dict(type="str", no_log=True),
+    )
+
+    module_args = dict(
+        name=dict(type="str"),
+        desc=dict(type="str"),
+        cluster=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive
+        ),
+        target_prefix=dict(type="str"),
+        volume_group_uuid=dict(type="str"),
+        flash_mode=dict(type="bool", default=False),
+        disks=dict(type="list", elements="dict", options=disk_spec),
+        vms=dict(
+            type="list",
+            elements="dict",
+            options=entity_by_spec,
+            mutually_exclusive=mutually_exclusive,
+        ),
+        load_balance=dict(type="bool", default=False),
+        clients=dict(
+            type="list",
+            elements="dict",
+            options=client_spec,
+            mutually_exclusive=[("uuid", "iscsi_iqn", "iscsi_ip")],
+        ),
+        CHAP_auth=dict(type="bool", default=False),
+        target_password=dict(type="str", no_log=True),
+    )
+
+    return module_args
+
+
+def create_volume_group(module, result):
+    volume_group = VolumeGroup(module)
+    spec, error = volume_group.get_spec()
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating volume_groups spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        result["response"]["disks"] = module.params.get("disks")
+        result["response"]["vms"] = module.params.get("vms")
+        result["response"]["clients"] = module.params.get("clients")
+        return
+
+    resp = volume_group.create(spec)
+    task_uuid = resp["data"]["extId"][-36:]
```

Comment:
If disks, vms or client creation is not required from user then there should be option for not waiting for volume groups create ?

Reply:
need to wait for VG uuid